osbuild/buildroot.py

```diff
@@ -80,7 +80,16 @@ def run(self, argv, binds=None, readonly_binds=None, **kwargs):
             f"--directory={self.root}",
             *[f"--bind={b}" for b in (binds or [])],
             *[f"--bind-ro={b}" for b in [f"{self.api}:/run/osbuild/api"] + (readonly_binds or [])],
-            ] + argv, check=check, **kwargs)
+            ]
+        try:
+            # systemd-nspawn changed the stdin handling, and added the --pipe option to restore the
+            # behavior we depend on. We need to be able to work on systemd with and without this
+            # patch.
+            ret = subprocess.run(args + ["--pipe"] + argv, check=True, **kwargs)
+        except subprocess.CalledProcessError:
+            ret = subprocess.run(args + argv, check=check, **kwargs)
```

Comment:
Alas, this means we'll run twice on "normal" failures as well.

Reply:
Check for `systemd-nspawn: unrecognized option '--pipe'`? :sob:

Reply:
We would have to capture the output in order to do it. That would completely change the way we see output from osbuild.

Reply:
Yeah, you are right, this patch is stupid. @gicmo, I was thinking of something like that, just lacked the right magic incantation of nspawn that would work correctly... I discussed offline with Lars and we may just rework the whole thing, but in the meantime I'll rebase your suggestion, as I sort of love it (despite it also sort of making me feel unwell).

Reply:
Meh, @gicmo's suggestion didn't work for me. We'll just rework it.